craft_parts/executor/part_handler.py

```diff
@@ -719,60 +767,65 @@ def _migrate_overlay_files_to_prime(self) -> None:
         of visible overlay entries, including OCI-compatible whiteout files and
         opaque directories.
         """
-        prime_overlay_state_path = states.get_overlay_migration_state_path(
-            self._part.overlay_dir, Step.PRIME
-        )
-
-        # Overlay data is migrated to prime only when the first part declaring overlay
-        # parameters is migrated.
-        if prime_overlay_state_path.exists():
-            logger.debug(
-                "prime overlay migration state exists, not migrating overlay data"
-            )
-            return
-
         parts_with_overlay = get_parts_with_overlay(part_list=self._part_list)
         if self._part not in parts_with_overlay:
             return
 
         logger.debug("priming overlay files")
-        migrated_files: set[str] = set()
-        migrated_dirs: set[str] = set()
-
-        # process layers from top to bottom (reversed)
```

Comment:
You lost this comment when you refactored this code, can you add it back?

Reply:
Added comment back.